NewsApp/app/src/main/java/com/example/newsapp/features/auth/login/LoginFragment.kt

```diff
@@ -4,17 +4,67 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.example.newsapp.R
 import com.example.newsapp.databinding.FragmentLoginBinding
+import com.example.newsapp.di.NewsManager
+import com.example.newsapp.features.auth.AuthViewModel
+import com.example.newsapp.features.auth.MyViewModelFactory
 
 class LoginFragment : Fragment() {
     private lateinit var binding: FragmentLoginBinding
 
+    private val viewModel: AuthViewModel by viewModels {
+        MyViewModelFactory(NewsManager.userRepo)
+    }
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         binding = FragmentLoginBinding.inflate(layoutInflater)
         return binding.root
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.loginBtn.setOnClickListener {
+            doLogin()
+        }
+
+        binding.registerBtn.setOnClickListener {
+            doRegister()
```

Comment:
`gotoRegister` / `navigateToRegister`?

we can also do the navigation in the MVVM annoying way lol.